File: app/src/main/java/com/stevenfrew/beatprompter/midi/NoValue.java
package com.stevenfrew.beatprompter.midi;

/**
 * Represents "no value", used for matching against nothing.
 */
public class NoValue extends Value {
    @Override
    byte resolve(byte[] arguments, byte channel) {
        return 0;
    }

    @Override
    boolean matches(Value otherValue) {
        return false;
    }

    @Override
    public String toString()
    {
        return "";
    }
}

File: app/src/main/kotlin/com/stevenfrew/beatprompter/midi/alias/NoValue.kt
package com.stevenfrew.beatprompter.midi.alias

/**
 * Represents "no value", used for matching against nothing.
 */
class NoValue : Value() {
	override fun resolve(arguments: ByteArray, channel: Byte): Byte = 0
	override fun matches(otherValue: Value?): Boolean = false
	override fun toString(): String = ""
}